charts/m4d/templates/cluster-metadata-config.yaml

```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-metadata
+data:
+  {{- if .Values.clusterMetadata.clusterName }}
```

Comment:
If name, region and zone are not defined the cluster managers will fail when trying to access these fields. 
The if statement suggests that it may be a property which value can be nil.

We should think about maybe setting default values for these properties. Or find a mechanism to fail helm if they are not set. (Otherwise the deployment will fail at some point and the user has to figure out where the error comes from by looking at pod logs)

Reply:
`required` (see https://helm.sh/docs/howto/charts_tips_and_tricks/)

Reply:
I was looking at the code and it seems like for local cluster mode it also reads the config map. It could also be changed to be hardcoded for local cluster mode but I prefer what you suggested to have a default for single cluster deployment.